### PR TITLE
Feat/sdk auth

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
## Description

This PR enables SDK authentication using SDK token value from environment value

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](SECURITY.md) is correct.
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/client/registry.ts [hunk 1]
```diff
@@ -1,9 +1,12 @@
 import { MapDocumentNode } from '@superfaceai/ast';
+import createDebug from 'debug';
 import * as zod from 'zod';
 
 import { isProviderJson, ProviderJson } from '../internal';
 import { HttpClient } from '../lib/http';
 
+const registryDebug = createDebug('superface:Registry');
```
Comment (lukas-valenta) [inline src/client/registry.ts:8]: i think all other are lowercase, let's keep it the same everywhere

File: src/client/registry.ts [hunk 2]
```diff
@@ -85,6 +88,26 @@ const bindResponseValidator = zod.object({
...
+  const tokenEnvName = 'SUPERFACE_SDK_TOKEN';
+  //Load superface token
+  const loadedToken = process.env[tokenEnvName];
+  if (!loadedToken) {
+    registryDebug(`Environment variable ${tokenEnvName} not found`)
+
+    return;
+  }
+  const token = loadedToken.trim();
+  const tokenRegexp = /^(sfs)_(.+)_([0-9A-F]{8}$)/i;
```
Comment (TheEdward162) [inline src/client/registry.ts:101]: Greedy regex parser goes brrr. Pls replace `.` with `[^_]` at least.

Comment (TheEdward162) [reply src/client/registry.ts:101]: Also the `$` doesn't need to be in the last group

Comment (Jakub-Vacek) [reply src/client/registry.ts:101]: I just reused regex from BE 🤨🤷

File: src/client/registry.ts (hunk 1 above)
Comment (Jakub-Vacek) [reply src/client/registry.ts:8]: At least on is upper cased https://github.com/superfaceai/one-sdk-js/blob/main/src/client/profile-provider.ts#L49
I will fix it to lower case everywhere :)

File: src/client/registry.ts [hunk 3]
```diff
@@ -98,7 +98,7 @@ export function loadSdkAuthToken(): string | undefined {
     return;
   }
   const token = loadedToken.trim();
-  const tokenRegexp = /^(sfs)_(.+)_([0-9A-F]{8}$)/i;
+  const tokenRegexp = /^(sfs)_([^_]+)_([0-9A-F]{8})/i;
```
Comment (TheEdward162) [inline src/client/registry.ts:101]: I meant to keep the `$` but to move it out of the group :)

**Review by TheEdward162 — APPROVED** [review]

**Pull request merged**